index.js

```diff
@@ -124,39 +126,68 @@ const headers = {
 				}));
 			}, []),
 
-			{ errorCount, warningCount } = report;
-
+			{
+				errorCount, warningCount
+			} = report;
 		return {
-			conclusion: errorCount > 0 ? 'failure' : 'success',
-			output: {
-				title: checkName,
-				summary: `${ errorCount } error(s), ${ warningCount } warning(s) found`,
-				text: 'A little bit of text',
-				annotations
-			}
+			annotations,
+			errorCount,
+			warningCount
 		};
 	},
-	updateCheck = async (id, conclusion, output) =>
+	updateCheck = async (id, opts = {}) =>
 		await request(`https://api.github.com/repos/${ owner }/${ repo }/check-runs/${ id }`, {
 			method: 'PATCH',
 			headers,
 			body: {
 				name: checkName,
 				head_sha: checkSha,
-				status: 'completed',
-				completed_at: (new Date()).toISOString(),
-				conclusion,
-				output
+				...opts
 			}
 		}),
+	updateChecks = async (id, {
+		errorCount, warningCount, annotations: all
+	}) => {
+		const chunkSize = 50,
+			chunksLength = Math.ceil(all.length / 50);
+		await Promise.all(new Array(chunksLength).fill().map((_, i) => {
+			const annotations = all.slice(i * chunkSize, (i + 1) * chunkSize);
+			return updateCheck(id, i < chunksLength - 1
+				? {
+					status: 'in_progress',
+					output: {
+						title: checkName,
+						summary: `${ errorCount } error(s), ${ warningCount } warning(s) found`,
+						annotations
+					}
+				}
+				: {
+					status: 'completed',
+					completed_at: (new Date()).toISOString(),
+					conclusion: errorCount > 0 ? 'failure' : 'success',
+					output: {
+						title: checkName,
+						summary: `${ errorCount } error(s), ${ warningCount } warning(s) found`,
+						annotations
+					}
+				}
+			);
+		}));
+	},
 	run = async () => {
 		const id = await createCheck();
 		try {
-			const { conclusion, output } = await eslint();
-			console.log(conclusion, output);
-			await updateCheck(id, conclusion, output);
+			await updateChecks(id, await eslint());
 		} catch (err) {
-			await updateCheck(id, 'failure');
+			await updateCheck(id, {
+				conclusion: 'failure',
+				status: 'completed',
+				completed_at: (new Date()).toISOString(),
+				output: {
+					title: checkName,
+					summary: 'Something went wrong while performing the check ...'
```

Comment:
Maybe include the err.message in the summary?

Reply:
Yes.